src/engineio/socket.rs

```diff
@@ -609,7 +611,7 @@ impl EngineSocket {
                     PacketId::Message => {
                         let on_data = self.on_data.read()?;
                         if let Some(function) = on_data.as_ref() {
-                            spawn_scoped!(function(packet.data));
+                            spawn_scoped!(function(packet.data.clone()));
```

Comment:
Mhh how comes we're needing a clone here? Couldn't we just use `into_iter` here? For that case it might also make sense to implement `Iter` for `Payload`.

Reply:
Done and done